src/components/calcite-ui-icon/calcite-ui-icon.tsx

```diff
@@ -0,0 +1,182 @@
+import {
+  Component,
+  h,
+  Element,
+  Prop,
+  State,
+  Watch,
+  Host,
+  Build
+} from "@stencil/core";
+import { CSS } from "./resources";
+import { getElementDir } from "../../utils/dom";
+import { fetchIcon, scaleToPx } from "./utils";
+import { Scale } from "../../interfaces/UIIcon";
+import { Theme } from "../../interfaces/common";
+
+@Component({
+  assetsDirs: ["assets"],
+  tag: "calcite-ui-icon",
+  styleUrl: "calcite-ui-icon.scss",
+  shadow: true
+})
+export class CalciteUIIcon {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element()
+  el: HTMLElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * When true, the icon will be filled.
+   */
+  @Prop({
+    reflect: true
+  })
+  filled: boolean = false;
+
+  /**
+   * The UI icon to display. The value of this property must match the icon name from https://esri.github.io/calcite-ui-icons/.
+   */
+  @Prop({
+    reflect: true
+  })
+  icon: string = null;
+
+  /**
+   * When true, the icon will be mirrored when the element direction is 'rtl'.
+   */
+  @Prop({
+    reflect: true
+  })
+  mirrored: boolean = false;
+
+  /**
+   * Icon scale. Can be "s" | "m" | "l".
+   */
+  @Prop({
+    reflect: true
+  })
+  scale: Scale = "m";
+
+  /**
+   * Icon theme. Can be "light" or "dark".
+   */
+  @Prop({
+    reflect: true
+  })
+  theme: Theme = "light";
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  connectedCallback(): void {
+    this.waitUntilVisible(() => {
+      this.visible = true;
+      this.loadIconPathData();
+    });
+  }
+
+  disconnectedCallback(): void {
+    if (this.intersectionObserver) {
+      this.intersectionObserver.disconnect();
+      this.intersectionObserver = null;
+    }
+  }
+
+  async componentWillLoad(): Promise<void> {
+    this.loadIconPathData();
+  }
+
+  render() {
+    const { el, mirrored, pathData, scale } = this;
+    const dir = getElementDir(el);
+    const size = scaleToPx[scale];
+
+    return (
+      <Host role="img">
```

Comment:
If we have an `img` role, do we also need an `aria-label`? Or are we expecting the user to add it? Should we add a label property?

Reply:
I was expecting the user to set it, but we could add that. Would `label` only be used to set `aria-label` or also `title`? Not sure if we want to keep those two separate. We can tackle this in a separate issue since users can set it themselves for now.

Reply:
Sometimes (ie. when the icon is used next to text) we should treat it as presentational. So `role="presentation"` or `alt=""` should be used. I'm not sure the implications of always having a `role=img` here. But something to be aware of.

Reply:
Sounds good. Maybe we have two props, one for a label and one for a title? Or we use the title for both?